openml/datasets/dataset.py

```diff
@@ -127,6 +130,10 @@ def __init__(self, name, description, format=None,
         self.name = name
         self.version = int(version) if version is not None else None
         self.description = description
+        if cache_format not in ['feather', 'pickle']:
+            raise ValueError("cache_format must be one of 'feather' or 'pickle'")
```

Comment:
Could you please also print the incorrect cache format? That sometimes makes it easier to debug if something fails.